### PR TITLE
Add android example

### PR DESCRIPTION
**Pull request by keith**
This is mostly mirrored from https://github.com/keith/bazel-rust-mobile-demo but serves as a test / example to validate changes pushed to this repo. It requires you have an Android NDK / SDK setup to run.

### CODE REVIEW

File: examples/android/WORKSPACE.bazel [hunk 1]
```diff
@@ -0,0 +1,50 @@
+workspace(name = "android_examples")
```
Comment (UebelAndre) [inline examples/android/WORKSPACE.bazel:1]: Does this need to be it's own workspace? I feel we'd get better coverage by having it be a part of `rules_rust_examples`

**Review by UebelAndre — CHANGES_REQUESTED** [review]
Thank you so much! One question but otherwise looks awesome to me 😄

File: examples/android/WORKSPACE.bazel (hunk 1 above)
Comment (keith) [reply examples/android/WORKSPACE.bazel:1]: the only annoyance is if you want to run those locally, you have to have the android SDK / NDK setup which I imagine most testers won't, I'm happy to merge them if you want though!

**Review by illicitonion — APPROVED** [review]
LGTM - thanks!

File: examples/android/WORKSPACE.bazel (hunk 1 above)
Comment (UebelAndre) [reply examples/android/WORKSPACE.bazel:1]: I think it can be done at a later date but in general, fewer workspaces would be better

**Review by UebelAndre — APPROVED** [review]

**Comment by UebelAndre** [comment]
@keith You wouldn't happen to have an iOS example project lying around would you 😄 (Thanks again!)

**Comment by keith** [comment]
Thanks! iOS example here https://github.com/bazelbuild/rules_rust/pull/1546

**Pull request merged**